i18n-helpers/src/lib.rs

```diff
@@ -120,4 +304,57 @@ mod tests {
             vec![(1, "This is\na paragraph.")]
         );
     }
+
+    #[test]
+    fn extract_msgs_styled_text() {
+        assert_eq!(
+            extract_msgs("**This** ~~message~~ _has_ *style*\n"),
+            vec![(1, "**This** ~~message~~ _has_ *style*")]
+        );
+    }
+
+    #[test]
+    fn extract_msgs_code_block() {
+        assert_eq!(
+            extract_msgs("Preamble\n```rust\nfn hello() {\n  todo!()\n}\n```\nPostamble"),
+            vec![
+                (1, "Preamble"),
+                (2, "```rust\nfn hello() {\n  todo!()\n}\n```"),
+                (7, "Postamble")
+            ],
+        );
+    }
+
+    #[test]
+    fn extract_msgs_details() {
+        assert_eq!(
+            // This isn't great, because the parser treats any data following a tag as also HTML,
```

Comment:
Yeah, I was worried about how the `details` elements would influence the Markdown parsing.

I had a plan to rework the speaker notes into a mdbook preprocessor instead. The idea would be to use Markdown syntax like

```markdown
# Speaker Notes {#speaker-notes}

Blah blah.
````

The `mdbook-xgettext` would extract this like normal and the translator would translate it (keeping the `speaker-notes` `id` attribute intact).

The new preprocessor could then run on the page and find the speaker notes and do the wrapping into `<details> ... </details>` as necessary.

I think that would simultaniously simplify the job of `mdbook-xgettext` and allow for a straight forward way to translate the heading. It would also mean less JavaScript code running on page load.

For now, I'm happy to rely on the pages having blank lines around the `details` tags.

Reply:
Oh, this plan would rely on [`ENABLE_HEADING_ATTRIBUTES`](https://docs.rs/pulldown-cmark/latest/pulldown_cmark/struct.Options.html#associatedconstant.ENABLE_HEADING_ATTRIBUTES) being set. It's not enabled as the moment by [`new_cmark_parser`](https://docs.rs/mdbook/latest/mdbook/utils/fn.new_cmark_parser.html) but I don't see why it shouldn't be enabled.

Reply:
I think that ENABLE_HEADING_ATTRIBUTES wouldn't matter for `mdboox-gettext`

Reply:
> I think that ENABLE_HEADING_ATTRIBUTES wouldn't matter for `mdboox-gettext`

Right, I agree — it would only matter for a new preprocessor which would use the ID to find the right tag to wrap.